src/color-scale/color-scale.js

```diff
@@ -53,17 +278,56 @@ const Self = class ColorScale extends ColorElement {
 
 		let i = 0;
 		let newSwatches = [];
-		for (let {name, color} of colors) {
+		for (let {name, color, intermediate} of colors) {
 			let swatch = this.#swatches[i] = this._el.swatches.children[i];
 
 			if (!swatch) {
 				this.#swatches[i] = swatch = document.createElement("color-swatch");
 				swatch.setAttribute("size", "large");
 				swatch.setAttribute("part", "color-swatch");
-				swatch.setAttribute("exportparts", "swatch, info, gamut");
+				swatch.setAttribute("exportparts", "swatch, info, gamut, label");
 				newSwatches.push(swatch);
 			}
 
+			// We should ignore all labelchange and colorchange events until the color scale is fully rendered
+			swatch.classList.add("ignore-updates");
+
+			swatch.classList[intermediate ? "add" : "remove"]("intermediate");
+
+			if (!intermediate) {
+				// Make the swatch editable, if needed
+				let editable;
+				if (this.editable?.name) {
+					(editable ??= {}).label = true;
+				}
+				if (this.editable?.color) {
+					(editable ??= {}).color = true;
+				}
+
+				if (editable) {
+					swatch.editable = editable;
+				}
+				else {
+					swatch.editable = false;
+				}
+
+				let deleteButton = swatch.querySelector("[part=delete-button]");
+				if (this.editable?.delete) {
+					if (!deleteButton) {
+						swatch.insertAdjacentHTML("beforeend", `
+							<button part="delete-button" title="Delete color">
+								<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
```

Comment:
Where does this icon come from? Also you don't need the SVG namespace for inline SVG.

Reply:
I took the icon from https://iconify.design/. I'll add a comment in the code.

> Also you don't need the SVG namespace for inline SVG.

Thanks. I'll fix it.